logicrunner/logicrunner.go

```diff
@@ -57,8 +57,20 @@ type Ref = insolar.Reference
 type ObjectState struct {
 	sync.Mutex
 
-	ExecutionState *ExecutionState
-	Validation     *ExecutionState
+	ExecutionBroker *ExecutionBroker
+	ExecutionState  *ExecutionState
+	Validation      *ExecutionState
+}
+
+func (st *ObjectState) InitAndGetExecution(lr *LogicRunner, ref *insolar.Reference) (*ExecutionState, *ExecutionBroker) {
```

Comment:
this is wrong. please make it "UpsertExecution" (or something like that) on StateStorage as it interface. Inject LR or anything else required to complete operation as dependency of StateStorage, don't pass it as argument into method.